lms/djangoapps/program_enrollments/api/linking.py

```diff
@@ -179,3 +181,38 @@ def link_program_enrollment_to_lms_user(program_enrollment, user):
             )
         )
         raise
+
+
+def _realize_course_access_roles(user, program_course_enrollment):
+    """
+    Convert any CourseAccessRoleAssignment objects, which represent pending CourseAccessRoles, into realized
+    CourseAccessRole objects as part of a program course enrollment.
+
+    Arguments:
+        user: User object for whom we are realizing CourseAccessRoles
+        program_course_enrollment: the ProgramCourseEnrollment object that represents the course the user
+            should be granted a CourseAccessRole in the context of
+    """
+    role_assignments = program_course_enrollment.courseaccessroleassignment_set.all()
+    program_enrollment = program_course_enrollment.program_enrollment
+
+    for role_assignment in role_assignments:
+        # currently, we only allow for an assignment of a "staff" role, but
+        # this allows us to expand the functionality to other roles, if need be
+        role = get_access_role_by_role_name(role_assignment.role)
+        role = role(program_course_enrollment.course_key)
```

Comment:
I don't like this variable re-assignment. Can we give a better name here? maybe `course_role_obj`?